[TEST] Failure in DeleteDataStreamRequestTests.testDeleteSnapshottingDataStream

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.action.admin.indices.datastream.DeleteDataStreamRequestTests.testDeleteSnapshottingDataStream" -Dtests.seed=65BD490888695D90 -Dtests.security.manager=true -Dtests.locale=fi-FI -Dtests.timezone=ECT -Druntime.java=15
```
```
org.elasticsearch.action.admin.indices.datastream.DeleteDataStreamRequestTests > testDeleteSnapshottingDataStream FAILED
    java.lang.AssertionError: Completed state must imply all shards completed but saw state [STARTED] and shards []
        at __randomizedtesting.SeedInfo.seed([65BD490888695D90:D53C2793DCBE4BFB]:0)
        at org.elasticsearch.cluster.SnapshotsInProgress$Entry.assertShardsConsistent(SnapshotsInProgress.java:252)
        at org.elasticsearch.cluster.SnapshotsInProgress$Entry.<init>(SnapshotsInProgress.java:191)
        at org.elasticsearch.cluster.SnapshotsInProgress$Entry.<init>(SnapshotsInProgress.java:163)
        at org.elasticsearch.action.admin.indices.datastream.DeleteDataStreamRequestTests.createEntry(DeleteDataStreamRequestTests.java:140)
        at org.elasticsearch.action.admin.indices.datastream.DeleteDataStreamRequestTests.testDeleteSnapshottingDataStream(DeleteDataStreamRequestTests.java:125)
```

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/datastream/DeleteDataStreamRequestTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/datastream/DeleteDataStreamRequestTests.java
@@ -136,7 +136,7 @@ public class DeleteDataStreamRequestTests extends AbstractWireSerializingTestCas
 
     private SnapshotsInProgress.Entry createEntry(String dataStreamName, String repo, boolean partial) {
         return new SnapshotsInProgress.Entry(new Snapshot(repo, new SnapshotId("", "")), false, partial,
-            SnapshotsInProgress.State.STARTED, Collections.emptyList(), Collections.singletonList(dataStreamName), 0, 1,
+            SnapshotsInProgress.State.SUCCESS, Collections.emptyList(), Collections.singletonList(dataStreamName), 0, 1,
             ImmutableOpenMap.of(), null, null, null);
     }
 

